components/shared/net/lib.rs

```diff
@@ -653,22 +657,43 @@ impl FetchThread {
                         .core_resource_thread
                         .send(CoreResourceMsg::Cancel(request_ids));
                 },
+                ToFetchThreadMessage::Exit => break,
             }
         }
     }
 }
 
 static FETCH_THREAD: OnceLock<Sender<ToFetchThreadMessage>> = OnceLock::new();
 
+/// Starts a fetch thread,
+/// and returns the join handle to it.
+pub fn start_fetch_thread(core_resource_thread: &CoreResourceThread) -> JoinHandle<()> {
+    let (sender, join_handle) = FetchThread::spawn(core_resource_thread);
+    FETCH_THREAD
+        .set(sender)
+        .expect("Fetch thread should be set only once on start-up");
+    join_handle
+}
+
+/// Send the exit message to the background thread,
+/// after which it can be joined on.
+pub fn exit_fetch_thread() {
+    let _ = FETCH_THREAD
+        .get()
+        .expect("Fetch thread should always be initialized on start-up")
+        .send(ToFetchThreadMessage::Exit);
+}
+
 /// Instruct the resource thread to make a new fetch request.
 pub fn fetch_async(
-    core_resource_thread: &CoreResourceThread,
+    _core_resource_thread: &CoreResourceThread,
```

Comment:
leaving this because it will be used for https://github.com/servo/servo/pull/38422